client/src/util/centroid.ts

```diff
@@ -16,22 +21,22 @@ import {
 
 /*
 Generates a mapping of labels to data needed to calculate centroids
-label -> {
- length: int,
- holdsFinite: Boolean,
- xCoordinates: Float32Array,
- yCoordinates: Float32Array
-}
 */
+
+export interface CentroidCoordinateObject {
+  length: number;
+  hasFinite: boolean;
+  xCoordinates: Float32Array;
+  yCoordinates: Float32Array;
+}
+
 const getCoordinatesByLabel = (
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  schema: any,
+  schema: Schema,
   categoryName: string,
   categoryDf: Dataframe,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  layoutChoice: any,
+  layoutChoice: LayoutChoiceState,
   layoutDf: Dataframe
-) => {
+): Map<string, CentroidCoordinateObject> => {
```

Comment:
we also support non-string labels.   Shouldn't this be typed:   `Map<LabelType, CentroidCoordinateObject>`?

Reply:
I wonder why I didn't get a type error for that, is it just because all of those types can be cast to string?

Reply:
you have it right.  A string is a LabelType, but the converse is not true.

Typing only helps you in some cases ;)